### PR TITLE
[nrf noup] Conditionally compile wpa commands help strings.

### PR DESCRIPTION
**Pull request by markaj-nordic**
This saves FLASH memory in cases when the wpa cli is not really used.

This patch essentially copies the `wpa_cli_cmd` array without help strings and pulls it
to the build process instead of the already existing set of "full" commands when the `CONFIG_WPA_CLI`
is not selected.

As a followup this could be refactored, e.g. help strings could be stored as an independent object, that could be
easily overwritten with NULLs (wpa_cli_cmd would then reference the strings by IDs etc.).

In the case of Matter if saves 0.7KB (enough to unblock nightly CI). For applications that utilize advanced supplicant features it's much more.

### CODE REVIEW

File: wpa_supplicant/wpa_cli_cmds.c [hunk 1]
```diff
@@ -3646,6 +3647,373 @@ static const struct wpa_cli_cmd wpa_cli_commands[] = {
 #endif  /* !CONFIG_ZEPHYR || (CONFIG_ZEPHYR && CONFIG_WPA_CLI)*/
 	{ NULL, NULL, NULL, cli_cmd_flag_none, NULL }
 };
+#else
+/* Compile commands without help string to spare some bytes */
+static const struct wpa_cli_cmd wpa_cli_commands[] = {
+	{"status", wpa_cli_cmd_status, NULL, cli_cmd_flag_none, NULL},
```
Comment (LuDuda) [inline wpa_supplicant/wpa_cli_cmds.c:3653]: Not sure what will be easier to maintain in future.. but alternatively you can do:

```
#ifdef CONFIG_WPA_CLI
#define HELP_MSG(msg) (msg)
#else
#define HELP_MSG(msg) NULL
#endif
```
And later:
```
{ "status", wpa_cli_cmd_status, NULL,
	  cli_cmd_flag_none,
	  HELP_MSG("[verbose] = get current WPA/EAPOL/EAP status") },
```

**Review by LuDuda — COMMENTED** [review]
Thanks :+1:

**Review by VivekUppunda — APPROVED** [review]

**Review by rado17 — APPROVED** [review]

**Comment by jfischer-no** [comment]
Really 700bytes of flash? How long will it take, one, two releases? Next matter revision update? Anyway, too fragile for me.

**Comment by markaj-nordic** [comment]
> Really 700bytes of flash? How long will it take, one, two releases? Next matter revision update? Anyway, too fragile for me.

Currently the Windows CI builds fail because of 20 bytes missing. And 700 is for Matter which uses only short set of wpa commands. I guess for more WiFi advanced apps, it's much more, yet it's only stripping off strings that are not used anyway. 

Don't worry, we will enable LTO for this application post 2.6.0. It will save ~50-60 KB.

File: wpa_supplicant/wpa_cli_cmds.c (hunk 1 above)
Comment (krish2718) [reply wpa_supplicant/wpa_cli_cmds.c:3653]: Yeah, using a macro to compile out RO strings is good. And this commands is `nrf noup` anyway, as I didn't want to duplicate the commands, so, moved to a new file `wpa_cli_cmds.c` which doesn't exist in the opensource, so, this is fine.

**Comment by PerMac** [comment]
@markaj-nordic 
> Don't worry, we will enable LTO for this application post 2.6.0. It will save ~50-60 KB.

Did that happen? Is this PR still needed?

**Comment by krish2718** [comment]
@markaj-nordic is this PR still needed? I guess not, so, can you please close?

**Comment by markaj-nordic** [comment]
> @markaj-nordic is this PR still needed? I guess not, so, can you please close?

Closed, but still I think I might be good to revisit this, especially in the context of samples that uses advanced features of the supplicant.

**Pull request closed without merging**